src/constr_gen.erl

```diff
@@ -4,7 +4,10 @@
 
 -export([
          gen_constrs_fun_group/1, gen_constrs_annotated_fun/2,
-         sanity_check/2
+         sanity_check/2,
+         % for tests
+         pat_guard_upper_lower/3,
+         ty_of_pat/3
         ]).
```

Comment:
If this is meant just for testing we could wrap it into an ifdef.

```suggestion
        ]).

-ifdef(TEST).
-export([
         pat_guard_upper_lower/3,
         ty_of_pat/3
        ]).
-endif.
```

Reply:
done